src/component/category-form/category-form.js

```diff
@@ -0,0 +1,82 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import autoBind from '../../utils/utils';
+import Modal from '../modal/modal';
+import './category-form.scss';
+
+const defaultState = {
+  title: '',
+  budget: '',
+  editing: false,
+};
+
+class CategoryForm extends React.Component {
+  constructor(props) {
+    super(props);
+
+    this.state = this.props.category || defaultState;
+    autoBind.call(this, CategoryForm);
+  }
+
+  handleSubmit(event) {
+    event.preventDefault();
+    this.setState({
+      editing: false,
+    });
+    this.props.onComplete(this.state);
+  }
```

Comment:
Might be good user experience to reset your state to default here so that the last thing a user entered doesn't remain in the input fields of the category form. When I deleted all my categories and created a new one, I still had the last values I had before in the inputs.